extensions/dirvish-emerge.el

```diff
@@ -369,19 +369,15 @@ If DEMOTE, shift them to the lowest instead."
   (cl-loop with (idx . files) = group
            with hide = (nth 2 (nth (1- idx) dirvish-emerge-groups))
            with desc = (nth 0 (nth (1- idx) dirvish-emerge-groups))
-           with start = (let* ((pos (point))
-                               (o (make-overlay pos pos)))
-                          (prog1 pos
-                            (when (and (not hide) (> (length files) 0))
-                              (overlay-put o 'dirvish-emerge-title t)
-                              (overlay-put o 'invisible t)
-                              (overlay-put o 'after-string
-                                           (dirvish-emerge--format-group-title desc)))))
+           with start = (point)
            for file in (reverse files) do (insert file "\n")
-           finally do (when hide
-                        (let ((o (make-overlay start (point))))
-                          (overlay-put o 'dirvish-emerge-hide t)
-                          (overlay-put o 'invisible t)))))
+           finally do (let ((o (make-overlay start (point))))
+                        (overlay-put o 'dirvish-emerge-group t)
+                        (overlay-put o 'before-string
+                                     (dirvish-emerge--format-group-title
+                                      (concat desc (when hide " (Hidden)"))))
+                        (overlay-put o 'invisible hide)
+                        (overlay-put o 'evaporate t))))
 
```

Comment:
> When I use evil-previous-line while stuck, I sometimes get evil-line-move: Beginning of buffer

I believe the problem lies in the the "\n" in the `before/after-string` overlay property.  A possible solution is insert the heading into the buffer.

```emacs-lisp
(defun dirvish-emerge--group-heading (desc hide)
  "Format group heading in Dirvish buffer.
DESC and HIDE are the group title and visibility respectively."
  (format "%s%s%s%s\n"
          (propertize " " 'font-lock-face
                      '(:inherit dirvish-emerge-group-title
                                 :strike-through t))
          (propertize (format " %s " (or desc "-"))
                      'face 'dirvish-emerge-group-title)
          (propertize (if hide "[...]" "")
                      'face 'dirvish-emerge-group-title)
          (propertize " " 'display '(space :align-to right)
                      'font-lock-face
                      '(:inherit dirvish-emerge-group-title
                                 :strike-through t))))

(defun dirvish-emerge--insert-group (group)
  "Insert GROUP to buffer."
  (pcase-let* ((`(,idx . ,files) group)
               (`(,desc _ ,hide) (nth (1- idx) dirvish-emerge-groups))
               (beg (progn (insert (dirvish-emerge--group-heading desc hide))
                           (point))))
    (dolist (file (reverse files)) (insert file "\n"))
    (let ((o (make-overlay beg (point))))
      (overlay-put o 'dirvish-emerge-group t)
      (overlay-put o 'evaporate t)
      (overlay-put o 'invisible hide))))
```

Reply:
I have tried inserting the heading into the buffer now. It solves this issue but creates a few others.